store/tikv/region_cache.go

```diff
@@ -66,17 +79,59 @@ func (c *RPCContext) GetStoreID() uint64 {
 	return 0
 }
 
+// DropInvalidRegion returns true if the region is invalid and then dropped or if it has been dropped
+func (c *RegionCache) DropInvalidRegion(id RegionVerID) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	cachedregion, ok := c.mu.regions[id]
+	if !ok {
+		return true
+	}
+	// we need to check ttl because concurrent threads may fresh it
+	// For example, T1 find ttl valid but before T1 freshes the ttl, T2 find ttl invalid
+	// In this case, T2 will invoke this function but T1 may have freshed the ttl
+	// Then, T2 should not drop the cached region
+	ttl := atomic.LoadInt64(&cachedregion.ttl) // we may not need an atomic load, I think a simple load is enough?
+	lastAccess := time.Unix(ttl, 0)
+	if time.Since(lastAccess) < rcDefaultRegionCacheTTL {
+		return false
+	}
+	delete(c.mu.regions, id)
+	return true
+}
+
+// GetCachedRegion returns a valid region
+func (c *RegionCache) GetCachedRegion(id RegionVerID) *Region {
+	for {
+		c.mu.RLock()
+		cachedregion, ok := c.mu.regions[id]
```

Comment:
We can just `RUnlock`  here.
The lock only protests the `regions` map.

Reply:
Yes, I will change the code. Thanks~